os/lib/memb.c

```diff
@@ -85,15 +84,13 @@ memb_free(struct memb *m, void *ptr)
      which the pointer "ptr" points to. */
   ptr2 = (char *)m->mem;
   for(i = 0; i < m->num; ++i) {
-    
     if(ptr2 == (char *)ptr) {
-      /* We've found to block to which "ptr" points so we decrease the
-	 reference count and return the new value of it. */
-      if(m->count[i] > 0) {
-	/* Make sure that we don't deallocate free memory. */
-	--(m->count[i]);
```

Comment:
It looks like in the original code there was a clear intention to omit the double-free bug.

Reply:
I do think that the original code might even have had the idea to allow calling alloc multiple times. But I do think your fix seems to clean up the code and make it more easy to read in general.

Reply:
I think you are right. I even had the impression that the original code had the intention to count references for some reason (the name count, ++/-- operations support this idea), which was unfinished and not really used. Such unfinished intentions can be dangerous, thank you for your corrections. As soon as time allows, we will try to verify the new version.